crates/ruff_cli/src/diagnostics.rs

```diff
@@ -146,18 +146,51 @@ impl AddAssign for Diagnostics {
     fn add_assign(&mut self, other: Self) {
         self.messages.extend(other.messages);
         self.imports.extend(other.imports);
-        for (filename, fixed) in other.fixed {
+        self.fixed += other.fixed;
+        self.notebook_indexes.extend(other.notebook_indexes);
+    }
+}
+
+/// A collection of fixes indexed by file path.
+#[derive(Debug, Default, PartialEq)]
+pub(crate) struct FixMap(FxHashMap<String, FixTable>);
+
+impl FixMap {
+    /// Returns `true` if there are no fixes in the map.
+    pub(crate) fn is_empty(&self) -> bool {
+        self.0.values().all(FixTable::is_empty)
+    }
```

Comment:
This is a somewhat expensive check for large fix tables. Can we instead avoid adding entries when they have no fixes or would that be confusing because it is no longer guaranteed that a key is present after inserting it.

Reply:
We can... It puts us back in the world we were in before (we'll need to take more care when creating these). I was hoping to capture hide that detail from callers. But I can change it.

Reply:
Can't we handle it inside of `from_iter`?

Reply:
Ah, right, I moved it.